test/unitTests/experimentalOmnisharpDownloader.test.ts

```diff
@@ -0,0 +1,172 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+import * as util from '../../src/common';
+import { should } from 'chai';
+import { Logger } from '../../src/logger';
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { ExperimentalOmnisharpDownloader } from '../../src/omnisharp/experimentalOmnisharp.Downloader';
+import { rimraf } from 'async-file';
+import { tmpdir } from 'os';
+
+const tmp = require('tmp');
+
+suite("DownloadAndInstallExperimentalVersion : Gets the version packages, downloads and installs them", () => {
+    let tmpDir = null;
+
+    setup(() => {
+        tmpDir = tmp.dirSync();
+        util.setExtensionPath(tmpDir.name);
+    });
+    
+    test('Packages are downloaded from the specified server url and installed at the specified path', async () => {
+        /* Download a test package that conatins a install_check_1.2.3.txt file and check whether the 
+           file appears at the expected path */
+        let version = "1.2.3";
+        let downloader = GetOmnisharpDownloader();
+        let serverUrl = "https://roslynomnisharp.blob.core.windows.net";
```

Comment:
How do we feel about unit tests that depend on the internet @TheRealPiotrP ?

Reply:
We tried using the mock server so we dont have to depend on the internet, but seems that the code creates an HTTPS request and we could not get the mock server running on the 443 port. Hence we have uploaded some dummy folders on the real server (like https://roslynomnisharp.blob.core.windows.net/releases/1.2.3/omnisharp-win-x64.zip) just for the testing purpose